### ISSUE TITLE
Install instructions do not end up in working environment 

### ISSUE THREAD

**Issue by bolkedebruin** [lyft/amundsenfrontendlibrary#6]
Hi Guys! Awesome that it is open sourced now! I was just trying things out and didn't get too far yet.

1. The linked metadata service is missing (you probably know this)

I also followed the instructions to a minimal view but got this

```
(venv) Bolkes-MBP:amundsenfrontendlibrary bolke$ python3 amundsen_application/wsgi.py
Registering post_exec action_log entry_point: logging_action_log = amundsen_application.log.action_log_callback:logging_action_log
2019-02-10T21:03:53+0100.686 [INFO] __init__.create_app:43 (54639:MainThread) - Created app with config name amundsen_application.config.LocalConfig
 * Serving Flask app "amundsen_application" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
2019-02-10T21:03:53+0100.748 [INFO] _internal._log:88 (54639:MainThread) -  * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
2019-02-10T21:04:08+0100.256 [ERROR] app.log_exception:1761 (54639:Thread-1) - Exception on / [GET]
Traceback (most recent call last):
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/lyft_amundsenfrontend-0.2.34-py3.7.egg/amundsen_application/api/__init__.py", line 16, in index
    return render_template("index.html")  # pragma: no cover
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/jinja2/environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/jinja2/loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/templating.py", line 58, in get_source
    return self._get_source_fast(environment, template)
  File "/Users/bolke/Documents/dev/amundsenfrontendlibrary/venv/lib/python3.7/site-packages/flask/templating.py", line 86, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: index.html
2019-02-10T21:04:08+0100.265 [INFO] _internal._log:88 (54639:Thread-1) - 127.0.0.1 - - [10/Feb/2019 21:04:08] "GET / HTTP/1.1" 500 -
2019-02-10T21:04:08+0100.421 [INFO] _internal._log:88 (54639:Thread-2) - 127.0.0.1 - - [10/Feb/2019 21:04:08] "GET /favicon.ico HTTP/1.1" 404 -
```

Like I said I didn't dive in yet.

**Comment by ttannis** [comment]
Hi @bolkedebruin, yes the metadata service is still missing. Thanks for raising the issue, I took a quick look and see the issue. We're on it.

**Comment by danwom** [comment]
@bolkedebruin Can you try running the following commands and then starting the server?
```
$ cd amundsen_application/static
$ npm install
$ npm run build
```

It appears the `amundsen_application/static/dist/templates/index.html` file could not be found. This file is created during the build process.

**Comment by feng-tao** [comment]
@bolkedebruin , great to see you here. We are still in the process of open source the final repo(amundsenmetadatalibrary). Will let you know once it is done.

**Comment by bolkedebruin** [comment]
@feng-tao no worries ;-) just making sure you guys know we are aware.

**Comment by bolkedebruin** [comment]
@danwom tried that again (it is part of the install instructions), but no luck yet.

**Comment by feng-tao** [comment]
@bolkedebruin , the metadata repo is now open source as well.(https://github.com/lyft/amundsenmetadatalibrary)

**Comment by ttannis** [comment]
@bolkedebruin do you have an `amundsen_application/static/dist` directory after running `npm run build`? If so what are the contents?

**Comment by jinhyukchang** [comment]
@bolkedebruin Are you still blocked?

**Comment by bolkedebruin** [comment]
blocked with the flu ;-)

**Comment by verdan** [comment]
got it working using install instructions. (with and without metadata and search services both) 👍

**Comment by feng-tao** [comment]
thanks @verdan for confirming :) and @bolkedebruin hopefully you will get well soon :)

**Comment by feng-tao** [comment]
@verdan @bolkedebruin we just create a slack workspace. Here is the invitation link(https://bit.ly/2tFsN1X). Feel free to ask us questions in that workspace :)

**Comment by ttannis** [comment]
@bolkedebruin hope you're feeling better. Please let us know if you are still having an issue or if we can go ahead and close this issue. I know @verdan got it working, but I'd like the rule out the chance that there is some environment specific issue with our setup instructions.

**Comment by bolkedebruin** [comment]
@ttanis thanks for checking up.on.me ;-) my team is working on it. They are figuring out some stuff now and see what the best approach is. Probably one or two extra ppl will get involved

**Comment by thenaturalist** [comment]
Hi, just followed the source installation guide (incl. `npm` builds) and I get the exact same error trace as posted initially. 

@ttannis contents of `amundsen_application/static/dist/` are
```288B Apr 10 12:57 .  
768B Apr 10 12:57 ..  
138K Apr 10 12:57 main.css. 
172K Apr 10 12:57 main.css.map
1.3M Apr 10 12:57 main.js
4.6M Apr 10 12:57 main.js.map
128B Apr 10 12:57 templates
102K Apr 10 12:57 vendors.js
237K Apr 10 12:57 vendors.js.map

**Comment by feng-tao** [comment]
cc @ttannis @danwom 
also @verdan , do you know if you encounter this issue or how you solve the issue on your end? My understanding is that we can't reproduce this issue, but it is worth to revisit it :(

**Comment by verdan** [comment]
Tried to setup the project from scratch, but still unable to reproduce the issue. 
The `/dist/` content looks good to me. @thenaturalist can you please let us know the contents of `amundsen_application/static/dist/templates/`? and also the version of Node and npm ?

**Comment by thenaturalist** [comment]
Versions:
- node: `11.13.0`
- npm: `6.7.0`

Contents for `amundsen_application/static/dist/templates`:
```
email.html
index.html
```

An error I didn't pay attention to the first time (terminal window size) is that `npm install` partially fails. Trace:

```
> fsevents@1.2.4 install /Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents
> node install

node-pre-gyp WARN Tried to download(404): https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.2.4/fse-v1.2.4-node-v67-darwin-x64.tar.gz
node-pre-gyp WARN Pre-built binaries not found for fsevents@1.2.4 and node@11.13.0 (node-v67 ABI, unknown) (falling back to source compile with node-gyp)
gyp ERR! configure error
gyp ERR! stack Error: Command failed: /Users/thenaturalist/git/amundsenfrontendlibrary/venv/bin/python -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:297:12)
gyp ERR! stack     at ChildProcess.emit (events.js:193:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:1001:16)
gyp ERR! stack     at Socket.stream.socket.on (internal/child_process.js:405:11)
gyp ERR! stack     at Socket.emit (events.js:193:13)
gyp ERR! stack     at Pipe._handle.close (net.js:614:12)
gyp ERR! System Darwin 18.5.0
gyp ERR! command "/usr/local/Cellar/node/11.13.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node" "--module_name=fse" "--module_path=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64" "--napi_version=4" "--node_abi_napi=napi"
gyp ERR! cwd /Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents
gyp ERR! node -v v11.13.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/Cellar/node/11.13.0/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node --module_name=fse --module_path=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64 --napi_version=4 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:193:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1001:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:266:5)
node-pre-gyp ERR! System Darwin 18.5.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/11.13.0/bin/node" "/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents
node-pre-gyp ERR! node -v v11.13.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.0
node-pre-gyp ERR! not ok
Failed to execute '/usr/local/Cellar/node/11.13.0/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node --module_name=fse --module_path=/Users/thenaturalist/git/amundsenfrontendlibrary/amundsen_application/static/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64 --napi_version=4 --node_abi_napi=napi' (1)
```

**Comment by ttannis** [comment]
@thenaturalist I think that the package in question is not supported by Node v11.x. Could you try downgrading your version of Node?
https://github.com/fsevents/fsevents/issues/246

**Comment by danwom** [comment]
For reference I am using `node v8.12.0` and `npm 6.4.1`. I haven't upgraded to `node v10.x.x` yet, so there may still be compatibility issues. I will update the doc with working versions.

**Comment by verdan** [comment]
> For reference I am using `node v8.12.0` and `npm 6.4.1`. I haven't upgraded to `node v10.x.x` yet, so there may still be compatibility issues. I will update the doc with working versions.

@danwom I'm running `node v10.12.0` and `npm 6.9.0`. We should also give node v11.x a try.

**Comment by thenaturalist** [comment]
@verdan Downgraded node to `v10.15.2` to no avail. Build works fine now, error stays exactly the same. Although I dabbled around a bit after reading up on the error, there is just no way `render_template('index.html')` completes successfully.

**Comment by verdan** [comment]
@thenaturalist please bare with us while we are still trying to reproduce this on our end. I tried to build the whole project from scratch without any problem. 
The only typo in the installation instruction was a requirements.txt https://github.com/lyft/amundsenfrontendlibrary/pull/75

Can you please confirm what's the output of `pip freeze`? Secondly, can you please try running `npm run dev-build` instead of `npm run build`?

**Comment by thenaturalist** [comment]
@verdan happy to. 

`npm run dev-build` leads to the same error. Again, `dist/*` is completely missing in the compiled `static` dir of the package in the venv. 

`pip freeze` output:
```amundsen-frontend==1.0.0
attrs==19.1.0
certifi==2019.3.9
chardet==3.0.4
Click==7.0
cookies==2.2.1
coverage==4.5.3
flake8==3.5.0
flake8-tidy-imports==1.1.0
Flask==1.0.2
idna==2.7
itsdangerous==1.1.0
Jinja2==2.10.1
MarkupSafe==1.1.1
marshmallow==2.15.3
mccabe==0.6.1
mock==2.0.0
more-itertools==7.0.0
mypy==0.660
mypy-extensions==0.4.1
pbr==5.1.3
pluggy==0.6.0
py==1.8.0
pycodestyle==2.3.1
pyflakes==1.6.0
pytest==3.5.1
pytest-cov==2.5.1
pytest-mock==1.1
requests==2.20.0
responses==0.9.0
six==1.12.0
typed-ast==1.2.0
typing==3.6.4
urllib3==1.24.1
Werkzeug==0.15.2```

**Comment by danwom** [comment]
I am fairly confident I have found a fix for this issue. Basically it is an order of operations error between `python3 setup.py install` and `npm run build`. The python install step generates an index of source files in `amundsen_frontend.egg-info/SOURCES.txt`. The `npm run build` step generates new files under `static/dist/` that are not yet indexed. When the server runs in the `python3 amundsen_application/wsgi.py` step, it doesn't know about the `static/dist/templates/index.html` file and fails.

TL;DR - Run `npm run build` before `python3 setup.py install`

**Comment by feng-tao** [comment]
cc @bolkedebruin  as well

**Comment by thenaturalist** [comment]
Can confirm this works now! Thanks guys!

**Comment by verdan** [comment]
cool @danwom @feng-tao. I believe now you also need to update the PyPi version, to make the `dist` available there.

**Comment by feng-tao** [comment]
thanks @verdan @thenaturalist @danwom @jinhyukchang @bolkedebruin , I also commit a pr to update the setup.py script per Verdan's suggestion and bump the pypi version. I think the issue could be closed now.
